docs/operations/batch-processing.md

```diff
@@ -0,0 +1,140 @@
+# Batch Processing
+
+In order to answer queries, JanusGraph has to perform queries
+against the storage backend.
+In general, there are two ways of doing this:
+
+- Once data from the backend is needed, execute a backend
+  query and continue with the result.
+- Maintain a list of what data is needed.
+  Once the list reaches a certain size, execute a batched
+  backend query to fetch all of it at once.
+  
+The first option tends to be more responsive and consume less
+memory because the query can emit the first results very early
+without waiting for larger batches of queries to complete.
+This is also the option that JanusGraph uses by default.
+The second option can be configured in multiple ways which are
+explained below.
+
+## Configuration Options
```

Comment:
nitpick: I feel this H2 heading is unnecessary. The H3 headings can be H2 headings.

Reply:
I'm fine with that